openassessment/xblock/submission_mixin.py

```diff
@@ -200,6 +189,56 @@ def save_submission(self, data, suffix=''):  # pylint: disable=unused-argument
         else:
             return {'success': False, 'msg': self._(u"This response was not submitted.")}
 
+    def get_saved_file_metadata(self):
```

Comment:
I would love (like, with the fury of a thousand suns) to have created (in a separate PR) a `file_upload_api.py` module that abstracts all of this away, and let's us treat file uploads as objects that have a simple interface.  Something like
```python
class FileUpload:
    def __init__(self, **kwargs):
        self.description = blah
        self.size = blah
        self.index = 3
        # etc.

    @classmethod
    def file_uploads_for_ora_block(cls, block):
        for thing in zipped_metadata:
            # etc.
```

I know I'm really waving hands here, hopefully you get the idea.  Would you be up for stubbing out a ticket and helping me groom this on Tuesday?
BTW, I think what you're doing here is a good first step.

Reply:
I would also love that, i'm in